tests/Testcontainers.Platform.Linux.Tests/DependsOnTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace Testcontainers.Tests;
+
+public sealed class DependsOnTests : IAsyncLifetime
+{
+    private const string DependsOnKey = "org.testcontainers.depends-on";
+
+    private const string DependsOnValue = "true";
+
+    private static readonly IDictionary<string, bool> LabelFilter = new Dictionary<string, bool> { { string.Join("=", DependsOnKey, DependsOnValue), true } };
+
+    private readonly IContainer _container = new ContainerBuilder()
+        .DependsOn(new ContainerBuilder()
+            .WithImage(CommonImages.Alpine)
+            .WithLabel(DependsOnKey, DependsOnValue)
+            .Build())
+        .DependsOn(new ContainerBuilder()
+            .WithImage(CommonImages.Alpine)
+            .WithLabel(DependsOnKey, DependsOnValue)
+            .Build())
+        .WithNetwork(new NetworkBuilder()
+            .WithLabel(DependsOnKey, DependsOnValue)
+            .Build())
+        .WithVolumeMount(new VolumeBuilder()
+            .WithLabel(DependsOnKey, DependsOnValue)
+            .Build(), "/workdir")
+        .WithImage(CommonImages.Alpine)
+        .WithLabel(DependsOnKey, DependsOnValue)
+        .Build();
+
+    public Task InitializeAsync()
+    {
+        return _container.StartAsync();
+    }
+
+    public Task DisposeAsync()
+    {
+        return _container.DisposeAsync().AsTask();
```

Comment:
@cristianrgreco @eddumelendez @mdelapenya @kiview Currently, dependent resources are not being removed when the parent container is disposed, as these resources might be shared across different configurations. Ryuk is responsible for cleaning them up.

My idea is to maintain an internal reference counter that increments with every create call. When the reference counter equals 1, we can remove the dependent resources as well (while disposing the parent). WDYT?

A general review would be appreciated. PR contains minor general improvements..